src/ale_interface.cpp

```diff
@@ -123,15 +178,44 @@ void ALEInterface::loadROM(string rom_file = "") {
     rom_file = theOSystem->romFile();
   }
   loadSettings(rom_file, theOSystem);
-  romSettings.reset(buildRomRLWrapper(rom_file));
+
+  RomSettings* romRlWrapper = buildRomRLWrapper(rom_file);
+  if (romRlWrapper == NULL) {
+    Logger::Info << "Unable to map rom '" << rom_file
+        << "'. Trying alternative method..." << std::endl;
+
+    std::string s = "Unknown";
+    std::string s1 = s.substr(0, s.find("("));
+    s1.erase(std::remove_if(s1.begin(), s1.end(), is_illegal), s1.end());
+    std::cout << s1 << std::endl;
+
+    const Properties properties = theOSystem->console().properties();
+    const std::string md5 = properties.get(Cartridge_MD5); //TODO there shouldn't be a need to do this twice
+    string rom_candidate = find_rom(md5);
+    if (rom_candidate != "") {
```

Comment:
or did you mean throw an exception instead of exit(1)? that part is not visible in the review comment